core/network/space.go

```diff
@@ -89,3 +95,44 @@ func (s SpaceInfos) GetByName(name string) *SpaceInfo {
 	}
 	return nil
 }
+
+var (
+	invalidSpaceNameChars = regexp.MustCompile("[^0-9a-z-]")
+	dashPrefix            = regexp.MustCompile("^-*")
+	dashSuffix            = regexp.MustCompile("-*$")
+	multipleDashes        = regexp.MustCompile("--+")
+)
+
+// ConvertSpaceName is used to massage provider-sourced (i.e. MAAS)
+// space names so that they conform to Juju's space name rules.
+func ConvertSpaceName(name string, existing set.Strings) string {
+	// Lower case and replace spaces with dashes.
+	name = strings.Replace(name, " ", "-", -1)
+	name = strings.ToLower(name)
+
+	// Remove any character not in the set "-", "a-z", "0-9".
+	name = invalidSpaceNameChars.ReplaceAllString(name, "")
```

Comment:
doesn't this mean that you can have 2 names that end up squashing to the same short name?
Squashing to "" seems pretty dangerous.

Reply:
I didn't change the logic of this method, but I agree that it is somewhat ill-conceived.

It would be much better to just error out for certain classes of names instead of trying to handle empty-string results and duplicates silently.